backend/danswer/connectors/danswer_jira/connector.py

```diff
@@ -0,0 +1,149 @@
+        raise ValueError("'projects' not found in the URL")
+
+    return jira_base, jira_project
+
+
+def fetch_jira_issues_batch(
+    jql: str,
+    start_index: int,
+    jira_client: JIRA,
+    batch_size: int = INDEX_BATCH_SIZE,
+) -> tuple[list[Document], int]:
+    doc_batch = []
+
+    batch = jira_client.search_issues(
+        jql,
+        startAt=start_index,
+        maxResults=batch_size,
+    )
+
+    for jira in batch:
+        if type(jira) != Issue:
+            logger.warning(f"Found Jira object not of type Issue {jira}")
+            continue
+
+        semantic_rep = (
+            f"Jira Ticket Summary: {jira.fields.summary}\n"
+            f"Description: {jira.fields.description}\n"
+            + "\n".join(
+                [f"Comment: {comment.body}" for comment in jira.fields.comment.comments]
+            )
+        )
+
+        page_url = f"{jira_client.client_info()}/browse/{jira.key}"
+
+        doc_batch.append(
+            Document(
+                id=page_url,
+                sections=[Section(link=page_url, text=semantic_rep)],
+                source=DocumentSource.CONFLUENCE,
+                semantic_identifier=jira.fields.summary,
+                metadata={},
+            )
+        )
+    return doc_batch, len(batch)
+
+
+class JiraConnector(LoadConnector):
+    def __init__(
+        self,
+        jira_project_url: str,
+        batch_size: int = INDEX_BATCH_SIZE,
+    ) -> None:
+        self.batch_size = batch_size
+        self.jira_base, self.jira_project = extract_jira_project(jira_project_url)
+        self.jira_client: JIRA | None = None
+
+    def load_credentials(self, credentials: dict[str, Any]) -> dict[str, Any] | None:
+        email = credentials["jira_user_email"]
+        api_token = credentials["jira_api_token"]
+        self.jira_client = JIRA(basic_auth=(email, api_token), server=self.jira_base)
+        return None
+
+    def load_from_state(self) -> GenerateDocumentsOutput:
+        if self.jira_client is None:
+            raise PermissionError(
+                "Jira Client is not set up, was load_credentials called?"
+            )
+
+        start_ind = 0
+        while True:
+            doc_batch, fetched_batch_size = fetch_jira_issues_batch(
+                f"project = {self.jira_project}",
+                start_ind,
+                self.jira_client,
+                self.batch_size,
+            )
+
+            if doc_batch:
+                yield doc_batch
+
+            start_ind += fetched_batch_size
+            if fetched_batch_size < self.batch_size:
+                break
+
+    def poll_source(
+        self, start: SecondsSinceUnixEpoch, end: SecondsSinceUnixEpoch
+    ) -> GenerateDocumentsOutput:
+        if self.jira_client is None:
+            raise PermissionError(
+                "Jira Client is not set up, was load_credentials called?"
+            )
+
+        start_datetime_str = datetime.fromtimestamp(start).strftime("%Y-%m-%dT%H:%M:%S")
```

Comment:
Not sure how this one works out... I guess the Jira server is timezone agnostic?